src/components/InputV2/Base.tsx

```diff
@@ -0,0 +1,101 @@
+import PropTypes from 'prop-types'
+import { Label } from './Label'
+
+export const BaseInput = ({
+  vertical,
+  centered,
+  label,
+  info,
+  error,
+  type,
+  autofocus,
+  disabled,
+  placeholder,
+  controlled,
+  value,
+  unit,
+  onChange,
+  onBlur
+}: BaseInputProps) => (
+  <Label disabled={disabled} vertical={vertical} centered={centered}>
+    {vertical ? (
+      <>
+        {label && <span>{label}</span>}
+        {info && <span className='info'>{info}</span>}
+      </>
+    ) : (
+      <div>
+        {label && <span>{label}</span>}
+        {info && <span className='info'>{info}</span>}
+        {error && <span className='error'>{error}</span>}
+      </div>
+    )}
+    <div className='text-input-wrapper'>
+      <input
+        className={error ? 'error' : ''}
+        type={type}
+        autoFocus={autofocus}
+        disabled={disabled}
+        placeholder={placeholder}
+        value={controlled ? value : undefined}
+        defaultValue={controlled ? undefined : value}
+        onChange={e => onChange(e.target.value)}
+        onBlur={e => onBlur(e.target.value)}
+        style={{ paddingRight: `${1 + 0.75 * unit.length}em` }}
+      />
+      {unit && <span className='unit'>{unit}</span>}
+    </div>
+    {vertical && error && <span className='error'>{error}</span>}
+  </Label>
+)
+
+export interface BaseInputProps {
```

Comment:
Hey I would make this a `Partial<` type, so anything rendering it doesn't get into typing trouble by not supplying a certain prop (same thing with default props not being considered by typescript compiler)